zk/utils/gas_limit.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+const (
+	preForkId7BlockGasLimit = 30_000_000
+	forkId7BlockGasLimit    = 18446744073709551615 // 0xffffffffffffffff
+	forkId8BlockGasLimit    = 1125899906842624     // 0x4000000000000
+)
+
+func GetExecutionGasLimit(forkId uint64) uint64 {
+	if forkId >= 8 {
+		return forkId8BlockGasLimit
+	}
+
+	// [hack] the rpc returns forkid8 value, but forkid7 is used in execution
```

Comment:
strange that Get**Execution**GasLimit returns a hacked value.

i would suggest to have a pair of methods, GetExecutionGasLimit and GetRPCGasLimit with appropriate logic.

Reply:
Looking at this code again more closely the comment doesn't make much sense any more about returning fork 7 for execution, because the code is doing the exact opposite.  Will have a scout around to see if we can make sense of this one

Reply:
yeah, looks fishy

I think forkid.7 executes with infinite (0xfff.fff) gasLimit, but for block RPC (and block hashes) returns the same as forkid.8 -- the network was rehashed for that.

Reply:
Definitely some history there, our sequencer also doesn't align with the RPC execution logic here.  Probably never surfaced because we haven't run a sequencer pre-etrog.  I have some code changes and simplification here, will leave Cardona overnight to make sure it doesn't break then can update this PR in the morning